app/components/preprint-status-banner/component.js

```diff
@@ -0,0 +1,119 @@
+import Ember from 'ember';
+const {
+    computed,
+} = Ember;
+
+const messagePendingPre = 'components.preprint-status-banner.message.pending_pre';
+const messagePendingPost ='components.preprint-status-banner.message.pending_post';
+const messageAccepted = 'components.preprint-status-banner.message.accepted';
+const messageRejected = 'components.preprint-status-banner.message.rejected';
+
+const statusPending = 'components.preprint-status-banner.pending';
+const statusAccepted = 'components.preprint-status-banner.accepted';
+const statusRejected = 'components.preprint-status-banner.rejected';
+
+const workflowPre = 'components.preprint-status-banner.pre_moderation';
+const workflowPost = 'components.preprint-status-banner.post_moderation';
+
+const iconPending = 'fa-hourglass-o';
+const iconAccepted = 'fa-check-circle-o';
+const iconRejected = 'fa-times-circle-o';
+
+export default Ember.Component.extend({
+    i18n: Ember.inject.service(),
+    theme: Ember.inject.service(),
+
+    // translations
+    labelModeratorFeedback: 'components.preprint-status-banner.feedback.moderator_feedback',
+    moderator: 'components.preprint-status-banner.feedback.moderator',
+    feedbackBaseMessage: 'components.preprint-status-banner.feedback.base',
+    baseMessage: 'components.preprint-status-banner.message.base',
+
+    classNames: ['preprint-status-component'],
+    classNameBindings: [
```

Comment:
I think having a single computed for classes (not sure if `classNames` can be a computed?) and just having a `switch` or `if`s would be nicer than all the `is___` computeds.